packages/lodestar/src/db/api/beacon/repositories/blockArchive.ts

```diff
@@ -29,4 +29,27 @@ export class BlockArchiveRepository extends BulkRepository<BeaconBlock> {
   public async add(value: BeaconBlock): Promise<void> {
     return this.set(value.slot, value);
   }
+
+  public async getAllBetween(
+    lowerLimit: number | null,
+    upperLimit: number | null,
+    step: number | null = null
+  ): Promise<BeaconBlock[]> {
+    const safeLowerLimit = lowerLimit || Buffer.alloc(0);
+    const safeUpperLimit = upperLimit || Number.MAX_SAFE_INTEGER;
+    const data = await this.db.search({
+      gt: encodeKey(this.bucket, safeLowerLimit),
+      lt: encodeKey(this.bucket, safeUpperLimit),
+    });
+    const processedData = (data || [])
+      .map((processedData) => deserialize(processedData, this.type))
```

Comment:
For the sake on typehinting, it is probably better to change it to deserialize<BeaconBlock>(...)